webapp/app/elster_client/elster_client.py

```diff
@@ -212,12 +212,16 @@ def _generate_est_request_data(form_data, year=2020):
     if adapted_form_data.get('steuernummer_exists') == 'no' and adapted_form_data.get('request_new_tax_number'):
         adapted_form_data['submission_without_tax_nr'] = True
 
-    if not current_user.is_active:
-        # no non-active user should come until here, but they should certainly not be able to send a tax
+    digitally_signed = current_user.is_authenticated and current_user.is_active
+
+    if not digitally_signed:
+        logger.warn('Elster_Client: User is not authenticated or active.')  
+        # no non-active user should come until here, but they should certainly not be able to send a tax        
         logout_user()
+        
```

Comment:
I think we would need to add a return here. Otherwise, we might still send the tax declaration with a wrong boolean value.

If we do not think that digitally_signed could be false, we do not need the guard at all.

Reply:
We can't just return here or? At least unit tests fail because the result of _generate_est_request_data is null. Maybe we should raise a error here?

Reply:
Yes, raising an error makes a lot of sense